### PR TITLE
Refactor bitesize8

### PR DESCRIPTION
**Pull request by t-young31**

### CODE REVIEW

**Comment by samcunliffe** [comment]
Is this targetted to the right branch? We're merging onto bitesize7 rather than chonk?

**Comment by t-young31** [comment]
yes and no! targeted 7 for a more reasonable diff. base should be automatically changed when 7 is merged 🤞🏼

**Comment by samcunliffe** [comment]
> yes and no! targeted 7 for a more reasonable diff. base should be automatically changed when 7 is merged 🤞🏼

TIL that github is smart.

File: tdms/include/simulation_parameters.h [hunk 1]
```diff
@@ -4,10 +4,12 @@ class SimulationParameters{
...
-    bool         is_multilayer = false;   // Is this simulation of a multilayer?
-    bool         is_disp_ml = false;      // Is the material dispersive?
+    double       omega_an      = 0.0;      // Angular ω
+    unsigned int Nsteps        = 0;        // Number of simulation steps
+    double       dt            = 0.0;      // Time step
+    bool         has_tdfdir    = false;    // Is the tdfdir (time domain field directory) defined?
+    bool         is_multilayer = false;    // Is this simulation of a multilayer?
+    bool         is_disp_ml    = false;    // Is the material dispersive?
+    double       to_l          = 0.0;      // time delay of pulse
+    double       hwhm          = 0.0;      // hwhm of pulse
```
Comment (samcunliffe) [inline tdms/include/simulation_parameters.h]: half width at half max?

File: tdms/include/simulation_parameters.h [hunk 2]
```diff
@@ -1,15 +1,28 @@
 
+struct PML{
+  int Dxl = 0;  // Thickness of lower pml in the x direction
+  int Dxu = 0;  //              upper
+  int Dyl = 0;
+  int Dyu = 0;
+  int Dzl = 0;
+  int Dzu = 0;
```
Comment (samcunliffe) [inline tdms/include/simulation_parameters.h:8]: Can we have `PerfectlyMatchedLayer` ?
If you don't like the long name (I understand and don't complain), then please docstring.

File: tdms/include/simulation_parameters.h [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
+    double       omega_an      = 0.0;      // Angular ω
+    int          Nt            = 0;        // Number of simulation steps
+    int          start_tind    = 0;        // Starting iteration number for the time steps
+    double       dt            = 0.0;      // Time step
+    bool         has_tdfdir    = false;    // Is the tdfdir (time domain field directory) defined?
+    bool         is_multilayer = false;    // Is this simulation of a multilayer?
+    bool         is_disp_ml    = false;    // Is the material dispersive?
+    double       to_l          = 0.0;      // time delay of pulse
+    double       hwhm          = 0.0;      // hwhm of pulse
+    PML          pml;                      // Perfectly matched layer
```
Comment (samcunliffe) [inline tdms/include/simulation_parameters.h]: Ah ok, well here it'd be
```
 double                hwhm          = 0.0;      // hwhm of pulse
 PerfectlyMatchedLayer pml;                      // Perfectly matched layer
```
which might be anoying to realign all of these things.

**Review by samcunliffe — APPROVED** [review]
Again. This all looks very nice to me.
8️⃣ ✅

File: tdms/include/simulation_parameters.h (hunk 2 above)
Comment (t-young31) [reply tdms/include/simulation_parameters.h:8]: we can and we shall – the alignment disadvantage is minimal!

**Pull request merged**